core/app/models/spree/payment.rb

```diff
@@ -9,6 +9,7 @@ class Payment < Spree::Base
 
     NON_RISKY_AVS_CODES = ['B', 'D', 'H', 'J', 'M', 'Q', 'T', 'V', 'X', 'Y'].freeze
     RISKY_AVS_CODES     = ['A', 'C', 'E', 'F', 'G', 'I', 'K', 'L', 'N', 'O', 'P', 'R', 'S', 'U', 'W', 'Z'].freeze
+    INOPERATIVE_STATES  = %w(failed invalid)
```

Comment:
@jitendra Yeah thats my point, 'invalid' state is made inoperative because we don't use it further, but it is not unsuccessful, because we never tried to capture it, so there is no point of being it unsuccessful or successful.